### ISSUE TITLE
SA1101 false positive with C# 14 extension members

### ISSUE THREAD

**Issue by AArnott** [DotNetAnalyzers/StyleCopAnalyzers#3954]
Given syntax like this:

```cs
public class MyClass
{
  public void Hi() { }
}

public static class MyExtensions
{
  extension(MyClass self)
  {
    public void Hey() => self.Hi();
  }
}
```

SA1101 mis-fires, wanting me to "Prefix local calls with this" where I use `self`. But `self` is not a field by C# syntax rules, and `this.` is invalid.
